mono/mini/driver.c

```diff
@@ -2006,6 +2020,8 @@ mono_main (int argc, char* argv[])
 	/* Set rootdir before loading config */
 	mono_set_rootdir ();
 
+	mini_threads_runtime_init ();
+
 	if (enable_profile) {
```

Comment:
You mean separate the profiler option parsing and profiler initialization, do the option stuff here, and the initialization in `mono_init_internal`?  I could give that a shot.